ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/PostExecutionConflictDetector.scala

```diff
@@ -11,16 +11,16 @@ import scala.concurrent.{ExecutionContext, Future}
   * An in-transaction post-execution conflict detection to be invoked as the last stage of a
   * pre-execution pipeline.
   */
-trait PostExecutionConflictDetector[StateKey, StateValue, ReadSet, WriteSet] {
+trait PostExecutionConflictDetector[+StateKey, -StateValue, -ReadSet, -WriteSet] {
```

Comment:
Can we possibly relax the variance requirements here?

Reply:
I can make `StateKey` and `StateValue` invariant. `WriteSet` needs to be contravariant to avoid an extra type parameter in the `EqualityBasedPostExecutionConflictDetector`, which ignores the write set entirely, and I figure `ReadSet` should then be covariant to match.